test/inductor/test_aot_inductor.py

```diff
@@ -1257,6 +1257,30 @@ def test_cond_non_tensor_predicates(self, dynamic):
             dynamic_shapes=dynamic_shapes,
         )
 
+    def test_cond_symint_input(self):
+        class M(torch.nn.Module):
+            def forward(self, x, y, z):
+                a = y.shape[0]
+                b = z.shape[0]
+
+                def true_fn(x):
+                    return x + a
+
+                def false_fn(x):
+                    return x + b * z
+
+                return torch.cond(x.shape[0] > 5, true_fn, false_fn, (x,))
+
+        input1 = (torch.ones(3, 3), torch.ones(5), torch.ones(3, 3))
+        input2 = (torch.ones(10, 3), torch.ones(6), torch.ones(10, 3))
+        inputs = (input1, input2)
+        dynamic_shapes = {"x": {0: Dim("d")}, "y": {0: Dim("d1")}, "z": {0: Dim("d")}}
+        self.check_model_with_multiple_inputs(
```

Comment:
I fixed a `check_model_with_multiple_inputs` bug in https://github.com/pytorch/pytorch/pull/138379/. After that, this test likely will fail the with_stack_allocation_and_minimal_arrayref_interface test. You can patch my change locally. If that's the case, you can simply do `fail_minimal_arrayref_interface(is_skip=True)` for this new test.

Reply:
Also make sure you land from fbcode. Otherwise it will give me a merge headache.

Reply:
Checked against your branch, looks like the tests are passing.